applications/serial_lte_modem/src/main.c

```diff
@@ -338,32 +339,73 @@ static void check_app_fota_status(void)
 int lte_auto_connect(void)
 {
 	int err = 0;
-
-#if defined(CONFIG_SLM_CARRIER)
+#if defined(CONFIG_SLM_AUTO_CONNECT)
 	int stat;
+	struct network_config {
+		/* Network-specific default system mode configuration by %XSYSTEMMODE */
```

Comment:
It would be good to point to XSYSTEMMODE documentation. Casual reader understands the first three options, but it is impossible to understand lte_preference without a comment or further documentation.

Reply:
Doc reference in code?

Reply:
If that is a no go, then a short explanation on the lte_preference options would be good.